### PR TITLE
Fake Xcode derivation

### PR DESCRIPTION
**Pull request by pikajude**
This makes building packages that expect an Xcode installation to exist both easier to do and easier to debug.

### CODE REVIEW

**Comment by domenkozar** [comment]
Could we keep such files outside of nixpkgs repository? It's already really huge.

File: lib/maintainers.nix [hunk 1]
```diff
@@ -152,7 +152,6 @@
   joachifm = "Joachim Fasting <joachifm@fastmail.fm>";
   joamaki = "Jussi Maki <joamaki@gmail.com>";
   joelmo = "Joel Moberg <joel.moberg@gmail.com>";
-  joelteon = "Joel Taylor <me@joelt.io>";
```
Comment (copumpkin) [inline lib/maintainers.nix]: I think you just broke hydra's evaluation of a bunch of other packages that refer to `joelteon`. Might want to fix those, too!

**Comment by copumpkin** [comment]
I'll take a closer look later, but @domenkozar which large files are you talking about?

**Comment by domenkozar** [comment]
@copumpkin nixpkgs is 450MB, let's make sure we don't add sourcecode to it. I'd just create a github repository and use fetchFromGithub to get sources.

**Comment by copumpkin** [comment]
You talking about the small shell scripts in this commit though? This
doesn't seem like a particularly bad offender.

I also share concerns about the repo size, but the giant autogenerated
package bundles seem like the bigger concern than one-off shell scripts. We
need to get import-from-derivation working properly and use it all over the
place!

**Comment by domenkozar** [comment]
@copumpkin well this is how code *starts* to grow around :)

**Comment by copumpkin** [comment]
@domenkozar sure, but I'd focus my shrinking efforts on the thing adding 99% of the new space automatically rather than the thing adding a kilobyte or two at a time, with human intervention. Humans are pretty bad at producing tons of data by hand. There's a cost to moving small scripts out of the repo (maintaining the hashes and so on), and when they're (small) core build scripts used by everything else, keeping them in the repo doesn't seem that bad to me. Or would you move cc-wrapper and the stdenv's setup.sh out of the repo too?

**Comment by domenkozar** [comment]
Sure, but expect me to comment on next commit that finds out this gluecode  isn't enough :)

**Comment by pikajude** [comment]
@copumpkin fixed

@domenkozar This derivation is 9.7KB in total and uses substituteAll pretty extensively, so my worry is that it wouldn't be as simple as it seems to break it out into its own package.

File: pkgs/development/tools/nix-xcode/setup-hook.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+fooPhase() {
```
Comment (spwhitt) [inline pkgs/development/tools/nix-xcode/setup-hook.sh]: Should we name this something like xcodePatchPhase?

**Comment by Mic92** [comment]
is this still relevant?

**Comment by periklis** [comment]
If we're going to merge this, following packages should be migrated to use `nix-code` as a dependency too:
Applications/audio/chuck/default.nix                                                                                                                                                                                                                                             

- [ ] [MacVim](/NixOS/nixpkgs/tree/master/pkgs/applications/editors/vim/macvim.nix#L33)                                                                                                                                                                                                                                          
- [ ] [QT 5.6](/NixOS/nixpkgs/tree/master/pkgs/development/libraries/qt-5/5.6/qtbase/default.nix#L80)                                                                                                                                                                                                                             
- [ ] [QT 5.7](/NixOS/nixpkgs/tree/master/pkgs/development/libraries/qt-5/5.7/qtbase/default.nix), currently in PR #23602                                                                                                                                                                                                                               
- [ ] [V8](/NixOS/nixpkgs/tree/master/pkgs/development/libraries/v8/3.16.14.nix#L27)                                                                                                                                                                                                                                             
- [ ] [Xcodewrapper?](/NixOS/nixpkgs/tree/master/pkgs/development/mobile/xcodeenv/xcodewrapper.nix#L8)                                                                                                                                                                                                                                    
- [ ] [Node Packages](/NixOS/nixpkgs/tree/master/pkgs/development/node-packages/default-v6.nix#L52)                                                                                                                                                                                                                                      
- [ ] [PhantomJS2](/NixOS/nixpkgs/tree/master/pkgs/development/tools/phantomjs2/default.nix#L66)                                                                                                                                                                                                                                     
- [ ] [Pinentry-mac](/NixOS/nixpkgs/tree/master/pkgs/tools/security/pinentry-mac/default.nix#L12)

**Comment by copumpkin** [comment]
I think @matthewbauer's `xcbuild` and nixpkgs-based SDK for it cover most of the use cases for this, right?

**Comment by copumpkin** [comment]
@wizeman @orivej another example

@pikajude you still interested in pushing this?

**Comment by mmahut** [comment]
Are there any updates on this pull request, please?

**Pull request closed without merging**